channeldb/graph.go

```diff
@@ -190,6 +190,36 @@ func (c *ChannelGraph) Database() *DB {
 	return c.db
 }
 
+// NumNodes returns the number of nodes currently in the graph
+func (c *ChannelGraph) NumNodes(tx *bbolt.Tx) (int, error) {
+
+	var result int
+	var err error
+
+	count := func(tx *bbolt.Tx) error {
+		nodes := tx.Bucket(nodeBucket)
+		if nodes == nil {
+			return ErrGraphNotFound
+		}
+
+		return nodes.ForEach(func(pubKey, _ []byte) error {
+			if len(pubKey) == 33 {
+				result += 1
+			}
+
+			return nil
+		})
+	}
+
+	if tx != nil {
+		err = count(tx)
+	} else {
+		err = c.db.View(count)
+	}
+
+	return result, err
```

Comment:
You could consider always returning `0, err` in case of an error. To prevent a more or less undefined result value.